### DISCUSSION TITLE
WeakReferenceMassenger Suggestion

### DISCUSSION THREAD

**Discussion by jrahma** [CommunityToolkit/Maui#846] [New Feature Discussions]
Hi,

Using WeakReferenceMessanger, why do I need to use the specific platform code just to show an alert?

If you see this Q&A, it’s Alor of code for iOS instead of simple DisplayAlert including the platform specific and the Action.

[https://learn.microsoft.com/en-us/answers/questions/1140607/weakreferencemessenger-not-triggered.html?childToView=1141676#answer-1141676
](https://learn.microsoft.com/en-us/answers/questions/1140607/weakreferencemessenger-not-triggered.html?childToView=1141676#answer-1141676
)

I hope the WeakReferenceMessenger will support that in future.

Thanks,
Jassim

**Comment by pictos** [comment]
Hey, @jrahma I don't believe this is an issue on `WeakReferenceMesssanger` or `CommunityToolkit.MAUI`, from the answer this looks to be a `Maui` issue I suggest you raise this question there.

Pinging @Sergio0694 just in case

**Comment by CliffAgius** [comment]
Closed and answered
